common/lib/util/eventemitter.ts

```diff
@@ -212,17 +214,15 @@ var EventEmitter = (function() {
 	 * @param currentState the name of the current state of this object
 	 * @param listener the listener to be called
 	 */
-	EventEmitter.prototype.whenState = function(targetState, currentState, listener /* ...listenerArgs */) {
-		var eventThis = {event:targetState},
-			self = this,
-			listenerArgs = Array.prototype.slice.call(arguments, 3);
+	private whenState(targetState: string, currentState: string, listener: Function, ...listenerArgs: unknown[] /* ...listenerArgs */) {
```

Comment:
Couple of things here:

- would prefer `listenerArguments` 😝 
- does the `/* ...listenerArgs */` comment need to remain?

Reply:
I'd rather keep it as `listenerArgs` given my comment about `args` being more idiomatic. As for the comment, 6f0aea0.